### ISSUE TITLE
Query language support for in array and exists

### ISSUE THREAD

**Issue by nwhatt** [balderdashy/waterline#1186]
In the following where clause I want to find all of my errors that are associated with an either a specific foo or a specific bar. 
```javascript
Errors.find()
.where(
  {or: [ {
    foo: fooIds},{
    bar: barIds}
]})
.populate('foo')
.populate('bar')
```
When fooIds is empty or barIds is empty I get all of Errors that have no foo set. 
One solution to this would be to design my db better, so that foo and bar are required on Error. 

I was hoping there might be a better way outside of redesign - maybe a way to specify something like 
```javascript
  {or: [ {
    foo: fooIds, foo: {'!': null},{
    bar: barIds, bar: {'!': null}
}]})
```
Meaning "foo in fooIds and foo not null." 

My current workaround is an if clause to dynamically construct the where clause based on whether or not my arrays are empty. Appreciate any suggestions - I may be missing a sweet query language feature.

**Comment by particlebanana** [comment]
Interesting, we actually don't have an `in` key in the query language. Let's change this to a feature request and add the ability to manually specify an `in` key. Would you submit a PR to the roadmap for this?

This would allow the following query to work:

```javascript
Errors.find({
  or: [
    { foo: { in: foodIds, '!': null } },
    { bar: { in: barIds, '!': null } }
  ]
}).exec(...)
```

**Comment by sailsbot** [comment]
Thanks for posting, @nwhatt.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Keeping this open until it is PRed into the roadmap.

**Comment by nwhatt** [comment]
Looks like it was merged by @particlebanana
https://github.com/balderdashy/waterline/pull/1195

**Comment by devinivy** [comment]
Great!  Thx!
